src/interfaces.ts

```diff
@@ -0,0 +1,8 @@
+type Method = 'POST' | 'GET' | 'PUT' | 'DELETE';
+export type Resource = 'create' | 'read' | 'change' | 'delete';
```

Comment:
Hmm, Why we need these? I think we can simplify logic with enums. https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
Changed.